libraries/AP_Airspeed/AP_Airspeed_SST_ND.cpp

```diff
@@ -0,0 +1,306 @@
+
+const float inH20_to_Pa = 249.08f;
+
+const uint8_t MN_ND210[8] = {0x4E, 0x44, 0x32, 0x31, 0x30, 0x00, 0x00, 0x00};
+const uint8_t MN_ND005D[8] = {0x4E, 0x44, 0x30, 0x30, 0x35, 0x44, 0x00, 0x00};
+const uint8_t MN_SST_ND[8] = {0x56, 0x4E, 0x31, 0x33, 0x31, 0x43, 0x00, 0x00};
+
+const float nd210_range[7] = {10.0, 5.0, 4.0, 2.0, 1.0, 0.5, 0.25}; // all in inH2O
+const float nd130_range[6] = {30.0, 20.0, 10.0, 5.0, 4.0, 2.0};
+const float nd160_range[8] = {60.0, 50.0, 40.0, 30.0, 20.0, 10.0, 5.0, 2.5};
+const float nd005d_range[6] = {138.4, 110.72, 55.36, 27.68, 22.14, 13.84}; // converted psi to inH2O
+const float vn131cm_range[8] = {23.6, 27.5, 31.5, 35.4, 39.4, 43.3, 47.2, 51.2}; //all in inH2O
+
+uint8_t config_setting[2] = {0x54, 0x00}; // notch filter disabled, bw limit set to 50Hz-> 148Hz odr with auto select, wdg disabled, pressure range set to 0b100
+uint8_t sst_config_setting[2] = {0x0A, 0x07}; //bw limit set to 50Hz -> 155.35Hz, pressure range set to 0b010
+
+AP_Airspeed_SST_ND::AP_Airspeed_SST_ND(AP_Airspeed &_frontend, uint8_t _instance) :
+    AP_Airspeed_Backend(_frontend, _instance)
+{
+}
+
+// probe for a sensor
+bool AP_Airspeed_SST_ND::probe(uint8_t bus, uint8_t address)
+{
+    _dev = hal.i2c_mgr->get_device(bus, address);
+    if (!_dev) {
+        return false;
+    }
+    WITH_SEMAPHORE(_dev->get_semaphore());
+
+    _dev->set_retries(20);
+    uint8_t reading[14]= {'\0'};
+    uint8_t model[8] = {'\0'};
+
+    if(!_dev->read(reading, 14)){
+        return false;
+    }else{
+        for (int i=0; i<14; i++){
+            if(i>5){
+                model[i-6]= reading[i];
+            }
+        }
+    }
+    //GCS_SEND_TEXT(MAV_SEVERITY_INFO,"Found bus %u addr 0x%02x", _dev->bus_num(), _dev->get_bus_address());
+    return matchModel(model);
+}
+
+// probe and initialise the sensor
+bool AP_Airspeed_SST_ND::init()
+{
+    static const uint8_t addresses[] = { ND_I2C_ADDR1, ND_I2C_ADDR2 };
+    if (bus_is_confgured()) {
+        for (uint8_t addr : addresses) {
+            if (probe(get_bus(), addr)) {
+                goto found_sensor;
+            }
+        }
+    } else {
+        FOREACH_I2C_EXTERNAL(bus) {
+            for (uint8_t addr : addresses) {
+                if (probe(bus, addr)) {
+                    goto found_sensor;
+                }
+            }
+        }
+        FOREACH_I2C_INTERNAL(bus) {
+            for (uint8_t addr : addresses) {
+                if (probe(bus, addr)) {
+                    goto found_sensor;
+                }
+            }
+        }
+    }
+
+    //GCS_SEND_TEXT(MAV_SEVERITY_ERROR, "SST_ND[%u]: no sensor found", get_instance());
+    return false;
+
+found_sensor:
+    _dev->set_device_type(uint8_t(DevType::SST_ND));
+    set_bus_id(_dev->get_bus_id());
+    //GCS_SEND_TEXT(MAV_SEVERITY_INFO,"SST_ND[%u]: Found bus %u addr 0x%02x", get_instance(), _dev->bus_num(), _dev->get_bus_address());
+
+    // send default configuration
+    WITH_SEMAPHORE(_dev->get_semaphore());
```

Comment:
Don't need this semaphore until the thread is started (i.e. not at all in here)

Reply:
Semaphore removed